### PR TITLE
CentOS/RH support, fetch dependencies from source repositories, other improvements 

### PR DESCRIPTION
**Pull request by hakamine**
- Fetch OS dependencies (osdeps .json files) from archivematica/
  archivematica-storage-service source repositories. osdeps files
  contain external repositories, repository keys, and package
  information. Is osdeps files are not found, it will install
  dependencies defined in the role as a fallback (only for Ubuntu 14)

- Add support for CentOS/RedHat 7 (osdeps required in the source repos),
  for CentOS/RedHat 7, archivematica services deployed with systemd.

- Instead of using distro packages, install pip with the get-pip.py
  script in order to have the most recent version. Similarly, install
  virtualenv with pip.

- Install each pipeline component (dashboard, MCP Server, MCP Client)
  in a different virtualenv

- Remove clamav installation tasks from this repo (move to a separate
  ansible role)

- Remove task that stops/disables apache2, to avoid errors when the
  apache package is not installed. This task was originally intended
  for the role to handle change from apache/mod_wsgi to nginx/gunicorn,
  from now on when switching apache must be manually disabled in
  advance (this shouldn't be a concern for most upgrades)

It currently can be tested by using branches dev/deployment-osdeps-ub-rh in the archivematica and storage service repositories (which contain the required osdeps files). 
Also ref. https://github.com/artefactual/deploy-pub/pull/26 , the Vagrantfiles there use this branch for provisioning (tested, confirmed to work)

### CODE REVIEW

File: tasks/pipeline-pipdeps-multi-venvs.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    name: "pip"
+    chdir: "{{ archivematica_src_dir }}/archivematica/src/dashboard"
+    virtualenv: "/usr/share/python/archivematica-dashboard"
+    state: "latest"
+
+
+- name: "Install common pip reqs in virtualenv archivematica-mcp-server"
+  pip:
+    chdir: "{{ archivematica_src_dir }}/archivematica/src/archivematicaCommon"
+    requirements: "{{ 'requirements/test.txt' if is_dev else 'requirements.txt'}}"
```
Comment (Hwesta) [inline tasks/pipeline-pipdeps-multi-venvs.yml:31]: Happy to see it installing the testing reqs if is_dev is set - I was going to implement this for AM but found it here first.

File: README.md [hunk 2]
```diff
@@ -136,17 +145,19 @@ It is also recommended to take backups of you system (Archivematica and Storage
      - role: "archivematica-src"
   vars:
        archivematica_src_dir: "/opt/archivematica"
-       archivematica_src_am_version: "stable/1.4.x"
-       archivematica_src_ss_version: "stable/0.7.x"
+       archivematica_src_am_version: "qa/1.6.x"
+       archivematica_src_ss_version: "qa/0.x"
```
Comment (Hwesta) [inline README.md]: We have stable branches now - stable/1.6.x & stable/0.10.x

File: README.md [hunk 3]
```diff
@@ -86,7 +86,9 @@ Role Variables
 - `archivematica_src_ssl_fullchain`:  (no default provided, shall be defined in the playbook or host_vars if SSL enabled)
 - `archivematica_src_ssl_privkey`: (no default provided, shall be defined in the playbook or host_vars if SSL enabled)
 
+### Pipeline venvs (added for 1.6.x)
 
+- `archivematica_src_am_multi_venvs`: create separate virtualenvs for each pipeline component (dashboard, MCPServer, MCPClient) (default:`no`)
```
Comment (Hwesta) [inline README.md:91]: Excited to see the virtualenv change - will make Python3 things easier.

File: README.md (hunk 2 above)
Comment (hakamine) [reply README.md]: Updated, thank you!

**Comment by hakamine** [comment]
For now, this PR will be merged to a newly created 'stable/1.6.x' branch instead of 'master', ref #105

**Pull request closed without merging**